Improved explenation around timing and added Graceful shutdown behaviour (#8114)
umbraco/UmbracoDocs: 17/umbraco-cms/develop-with-umbraco/configuration/distributedjobssettings.md

**Default:** `00:05:00` (5 minutes)

Adds a grace period on top of the job's own `Period` before a running job is considered stale.

A job that is marked as running in the database is normally skipped by other servers, which prevents duplicate execution. A job becomes eligible for recovery only when more than `Period + MaximumExecutionTime` has elapsed since `LastAttemptedRun`. The `Period` here is the job's own `Period` property — not the polling `Period` defined above.

For example, a job with a 20-minute `Period` and the default 5-minute `MaximumExecutionTime` is recoverable 25 minutes after the last attempted start.

> `MaximumExecutionTime` only applies when a job's `finally` block never runs. For example, when the server crashes or is forcibly killed mid-job. When a server shuts down gracefully, and the job observes the cancellation token, the host clears the running flag and stamps `LastRun` as part of the cleanup. The job is then eligible again on its normal cadence — `Period` after the cancellation moment — and `MaximumExecutionTime` is not used.

Reviewer: ```suggestion
`MaximumExecutionTime` only applies when a job's `finally` block never runs. For example, when the server crashes or is forcibly killed mid-job. Upon graceful shutdown, if the job observes the cancellation token, the host clears the running flag and stamps `LastRun` during cleanup. The job is then eligible again on its normal cadence — `Period` after the cancellation moment — and `MaximumExecutionTime` is not used.
```
